### ISSUE TITLE
Functions from other modules are never inlined

### ISSUE THREAD

**Issue by rniii** [melange-re/melange#1065]
As an example, here's a function composition operator:
```ocaml
let (>>) f g x = f (g x)
```

Using it from the same file as it is defined always gets the function inlined --
```ocaml
let logPlusTwo = Js.log >> ((+) 2)
```
```js
function logPlusTwo(param) {
  console.log(2 + param | 0);
}
```

But when moving the operator to something like an `Utils` module, the inlining never happens, which makes the output unnecessarily large. Explicitly annotating it with `[@inline]` also doesn't seem to do anything.

Is this a limitation with the build system working file-by-file? If so, is there any workaround?

**Comment by anmonteiro** [comment]
> Is this a limitation with the build system working file-by-file? If so, is there any workaround?

Yeah, effectively. We have limited, experimental support for cross-module optimizations. Try adding this in your dune file:

```lisp
({melange.,}compile_flags :standard --mel-cross-module-opt)
```

**Comment by anmonteiro** [comment]
I added a fairly comprehensive test for this in https://github.com/melange-re/melange/pull/1291. We could make a few improvements:

- perform compiler optimizations after inlining the cross-module function (would likely eliminate some of the calls to `Curry`
- store the relevant information in the `.cmj` even if `--mel-cross-module-opt` isn't passed in, such that libraries don't need to explicitly pass that flag (and optimizations can still be had in `melange.emit`)
